sympy/polys/matrices/domainmatrix.py

```diff
@@ -49,21 +69,98 @@ def from_list_sympy(cls, nrows, ncols, rows, **kwargs):
 
     @classmethod
     def from_Matrix(cls, M, **kwargs):
+        """Converts a Matrix to Domain Matrix of
+        suitable Domain
+
+        Examples
+        ========
+        Define a Domain Matrix on the field of Rationals
+
+        >>> from sympy import Matrix, Rational
+        >>> from sympy.polys.matrices import DomainMatrix
+        >>> A = DomainMatrix.from_Matrix(Matrix([
+        ...     [Rational(1,2), Rational(3,4)],
+        ...     [2, Rational(1,5)],
+        ... ]))
+        >>> A
+        DomainMatrix([[1/2, 3/4], [2, 1/5]], (2, 2), QQ)
+
+        Define a Domain matrix on the field of Reals.
+        >>> from sympy import Matrix
+        >>> from sympy.polys.matrices import DomainMatrix
+        >>> A = DomainMatrix.from_Matrix(Matrix([
+        ...     [1.0, 3.4],
+        ...     [2.4, 1]
+        ... ]))
+        >>> A
+        DomainMatrix([[1.0, 3.4], [2.4, 1.0]], (2, 2), RR)
+
+        """
         return cls.from_list_sympy(*M.shape, M.tolist(), **kwargs)
 
     @classmethod
     def get_domain(cls, items_sympy, **kwargs):
+
         K, items_K = construct_domain(items_sympy, **kwargs)
         return K, items_K
 
     def convert_to(self, K):
+        """ Converts an object of a given Domain to another specified
+        domain K
+
+        Parameters
+        ==========
+
+        K: Matrix is to be converted to Domain K
+
+        Examples
+        ========
+
+        >>> from sympy.polys.matrices import DomainMatrix
+        >>> from sympy import QQ, RR
+        >>> A = DomainMatrix([[QQ(1,2), 3]], (1, 2), QQ)
+        >>> A.convert_to(RR)
+        DomainMatrix([[0.5, 3.0]], (1, 2), RR)
+
+        >>> from sympy.polys.matrices import DomainMatrix
+        >>> from sympy import QQ, RR
+        >>> A = DomainMatrix([[RR(1.5), RR(1)], [RR(1.2), RR(0.3)]], (2, 2), RR)
+        >>> A.convert_to(QQ)
+        DomainMatrix([[3/2, 1], [6/5, 3/10]], (2, 2), QQ)
+
+        """
         return self.from_rep(self.rep.convert_to(K))
 
     def to_field(self):
         K = self.domain.get_field()
         return self.convert_to(K)
 
     def unify(self, other):
+        """Unifies the domain of 2 Matrices, so that
+        they both belong to a common unified Domain
+
+        Parameters
+        ==========
+
+        other: other Domain Matrix to be unified
+
+        Returns
+        =======
+
+        Two Matrices that contain the elements of the input
+        matrices but they now belong to a unified Domain
+
+        Examples
+        ========
+
+        >>> from sympy.polys.matrices import DomainMatrix
+        >>> from sympy import ZZ, RR
+        >>> A = DomainMatrix([[1, 2, 3]], (1, 3), ZZ)
+        >>> B = DomainMatrix([[RR(1.2), RR(2.0), RR(3.4)]], (1, 3), RR)
```

Comment:
I think it is best to avoid using RR as an example domain because DomainMatrix is not well implemented for RR. None of the algorithms makes any attempt to handle floating point difficulties. A better example here would be QQ.

Ideally the RR implementation of DomainMatrix should probably defer all of its calculations to mpmath's matrix class.

It would be good to have a see also here for the `Domain.unify` method.

Reply:
Ok, I'll fix the examples and add a new commit.